scripts/properly_populate_transaction_fees.js

```diff
@@ -0,0 +1,281 @@
+/**
+ * These are some recommendations for using the script in
+ * *development* mode. We currently have ~70000 transactions in our
+ * database, running it with a large batch to process all the
+ * transactions at once is the fastest way to run it and get the
+ * output, however it also exhausts node's memory limits with batches
+ * larger than ~50000. Use --max-old-space=8192 to do it all in one
+ * batch and iterate faster.
+ *
+ * For production usage, It might be a good idea to break it down in
+ * multiple batches to leave some room for the database to process
+ * operations from the web clients. I've been running 7 batches of
+ * 10000 and didn't take more than ~20s in my machine (i7~2.2GHz),
+ * which is just a little bit slower than running it all in one batch.
+ */
+import { ArgumentParser } from 'argparse';
+import models, { sequelize } from '../server/models';
+import * as transactionsLib from '../server/lib/transactions';
+import * as paymentsLib from '../server/lib/payments';
+import { OC_FEE_PERCENT } from '../server/constants/transactions';
+
+class Migration {
+  constructor(options) {
+    this.options = options;
+    this.offset = 0;
+    this.migrated = 0;
+  }
+
+  /** Retrieve the total number of valid transactions */
+  countValidTransactions = async () => {
+    return models.Transaction.count({ where: { deletedAt: null } });
+  }
+
+  /** Retrieve a batch of valid transactions */
+  retrieveValidTransactions = async () => {
+    const transactions = await models.Transaction.findAll({
+      where: { deletedAt: null },
+      order: ['TransactionGroup'],
+      limit: this.options.batchSize,
+      offset: this.offset,
+      include: [{ model: models.Collective, as: 'collective' }]
+    });
+    this.offset += transactions.length;
+    return transactions;
+  }
+
+  /** Convert `value` to negative if it's possitive */
+  toNegative = (value) => value > 0 ? -value : value;
+
+  /** Ensure that `tr` has the `hostCurrencyFxRate` field filled in */
+  ensureHostCurrencyFxRate = (tr) => {
+    if (tr.amount === tr.amountInHostCurrency && !tr.hostCurrencyFxRate)
+      tr.hostCurrencyFxRate = 1;
+  }
+
+  /** Return true if the transaction has any host fees */
+  hasHostFee = (tr) => tr.hostFeeInHostCurrency
+    && parseInt(tr.hostFeeInHostCurrency, 10) !== 0;
+  /** Return true if the transaction has any platform fees */
+  hasPlatformFee = (tr) => tr.paymentProcessorFeeInHostCurrency
+    && parseInt(tr.paymentProcessorFeeInHostCurrency, 10) !== 0;
+  /** Return false if there are no fees on a transaction */
+  hasFees = (tr) => this.hasHostFee(tr) || this.hasPlatformFee(tr)
+    || (tr.platformFeeInHostCurrency && parseInt(tr.platformFeeInHostCurrency, 10) !== 0);
+
+  rewriteFees = (credit, debit) => {
```

Comment:
Same comment here. I think `getFees` would be cleaner. It returns the fees and the caller is clearly responsible for setting and storing them. 

Since this is just a script, it's fine to leave it as it is. More a comment for future reusable components.

Reply:
I wrote an answer to the previous question that might address this too.